src/types.ts

```diff
@@ -72,3 +145,16 @@ export type Amounts = {
     beihilfebetrag: number
     mehrwertsteuerbetrag: number
 }
+
+export type GroupInvoiceByRecipientMethod = (
+    invoice: Invoice,
+    findRecipient: (
+        pflegekasseIK: string,
+        leistungsart: Leistungsart,
+        location: CareProviderLocationSchluessel,
+    ) => {
+        key: string,
+        recipient?: Recipient,
+        kostentraegerIK?: string,
+    }
+) => Record<string, { recipient?: Recipient, invoice: Invoice }>;
```

Comment:
FYI: As a non-typescript hack, I find this a bit confusing. (It is a function that creates a type?) Maybe a documentation comment would be helpful here.

Reply:
It's a method signature, it does not return a type, it is a function type.
I need it because it has a extensive method signature and there will be probably two functions that need to conform to this exact method signature. I will add a comment.